### ISSUE TITLE
refactor setup scripts

### ISSUE THREAD

**Issue by niphlod** [web2py/web2py#1078]
todo list:
- linux
   - support fedora and ubuntu only
   - a single setup script for each distro, with a parameter for additionally installing and setting up postgresql. Provide nginx + uwsgi setup, http + https sites holding web2py apps as content served through the root
- windows
   - support WS2012R2, setup http site holding web2py apps as content server through the root
- refactor the online book to promote nginx + uwsgi over apache
- change all password requests with getpass to mask the admin password

**Comment by mdipierro** [comment]
any news about this?

**Comment by pembeci** [comment]
What is the rationale of this part: *promote nginx + uwsgi over apache*? Is it because of general pros/cons of nginx + uwsgi over apache or are there any particular reasons like web2py's architecture being more suitable to the nginx option? Thanks.

**Comment by mdipierro** [comment]
It is not a specific web2py issue. Apache is a process driven architecture and it creates new processed to handle more requests. This means more memory consumption than nginx. Nginx achieves multitasking via non-blocking event-driven architecture and a single process.

    http://www.thegeekstuff.com/2013/11/nginx-vs-apache/

Here are some PHP benchmarks. We found similar results with Python frameworks.

   http://www.theorganicagency.com/apache-vs-nginx-performance-comparison/

Simply put, Nginx is a much more modern and lean system. Digital Ocean has a much less biased comparison that outlines the pros and cons of each: 

   https://www.digitalocean.com/community/tutorials/apache-vs-nginx-practical-considerations

**Comment by pembeci** [comment]
Thanks for the detailed answer. Especially, Digital Ocean's article is very extensive and fair.

**Comment by niphlod** [comment]
to me it's quite a totally separated issue, but the end-game is the same. IMHO uwsgi is a far superior way to handle any python web app over apache.

**Comment by BuhtigithuB** [comment]
There is also the aspect of making scripts more managable too I think...
Having all differents scripts to maintain is a PITA because they were not
DRY for simplicity...

Thanks to Simone to make them more maintainable even if you will lack all
the diversity that were there... I think we will be better server like
that, since the available scripts will be up to date...

:)

**Comment by mdipierro** [comment]
I think we should do this now rather than later. Simone, can you discuss over phone this week?

**Comment by BuhtigithuB** [comment]
@mdipierro and @niphlod : Is this a milestone for next release?

**Comment by mdipierro** [comment]
I have deleted lots of junk in scripts already. About the remaining scripts, I say we should delete them if they do not work. Otherwise keep them. This should be addressed not in the next but in the release fater the next.

**Comment by mdipierro** [comment]
many scripts have been deleted. I believe we already use getpass everywhere. I do not think we should maintain only fedora ubuntu. If people send other scripts and they work there is no reason to discard them.

**Comment by BuhtigithuB** [comment]
Can we consider this one done?

**Comment by niphlod** [comment]
Did you try alla leftover scripts and confirmed they work ad expected? The todolist is at the top and IMHO nothing has ben done
